src/base/bittorrent/torrenthandle.cpp

```diff
@@ -487,6 +487,7 @@ bool TorrentHandle::needSaveResumeData() const
 
 void TorrentHandle::saveResumeData()
 {
+    updateStatus(); // to update queue_position, see discussion in PR #6154
```

Comment:
I was thinking of explicitly asking for new status only during shutdown. With an extra call to explicitly update the status just before we call `TorrentHandle::saveResumeData()`.